packages/ui/src/components/va-chip/VaChip.vue

```diff
@@ -35,139 +35,136 @@
         class="va-chip__close-icon"
         name="close"
         :size="iconSize"
-        @click.stop="close()"
+        @click.stop="close"
       />
     </span>
   </component>
 </template>
 
 <script lang="ts">
-import { watch } from 'vue'
-import { Options, prop, mixins, setup, Vue } from 'vue-class-component'
-
+import { defineComponent, PropType, ref, computed } from 'vue'
 import {
   getBoxShadowColor,
   getHoverColor,
-  getFocusColor, getTextColor,
+  getFocusColor,
+  getTextColor,
 } from '../../services/color-config/color-functions'
-import ColorMixin from '../../services/color-config/ColorMixin'
-import { RouterLinkMixin } from '../../mixins/RouterLinkMixin/RouterLinkMixin'
-import { StatefulMixin } from '../../mixins/StatefulMixin/StatefulMixin'
+import { useRouterLink, useRouterLinkProps } from '../../composables/useRouterLink'
 import useKeyboardOnlyFocus from '../../composables/useKeyboardOnlyFocus'
+import { useColors, useColorProps } from '../../composables/useColor'
+import { useStateful, useStatefulEmits, useStatefulProps } from '../../composables/useStateful'
 import VaIcon from '../va-icon'
 
-class ChipProps {
-  modelValue = prop<boolean>({ type: Boolean, default: true })
-  closeable = prop<boolean>({ type: Boolean, default: false })
-  color = prop<string>({ type: String, default: 'primary' })
-  outline = prop<boolean>({ type: Boolean, default: false })
-  flat = prop<boolean>({ type: Boolean, default: false })
-  icon = prop<string>({ type: String, default: '' })
-  disabled = prop<boolean>({ type: Boolean, default: false })
-  square = prop<boolean>({ type: Boolean, default: false })
-  tag = prop<string>({ type: String, default: 'span' })
-  size = prop<string>({
-    type: String,
-    default: 'medium',
-    validator: (value: string) => {
-      return ['medium', 'small', 'large'].includes(value)
-    },
-  })
-  shadow = prop<boolean>({ type: Boolean, default: false })
-}
-
-const ChipPropsMixin = Vue.with(ChipProps)
-
-@Options({
+export default defineComponent({
   name: 'VaChip',
+
   components: { VaIcon },
-  emits: ['focus'],
-})
-export default class VaChip extends mixins(
-  RouterLinkMixin,
-  StatefulMixin,
-  ColorMixin,
-  ChipPropsMixin,
-) {
-  SetupContext = setup(() => {
+
+  emits: [...useStatefulEmits, 'focus'],
+
+  props: {
+    ...useRouterLinkProps,
+    ...useColorProps,
+    ...useStatefulProps,
+    modelValue: { type: Boolean as PropType<boolean>, default: true },
```

Comment:
```suggestion
    modelValue: { type: Boolean, default: true },
```

Vue will cast type from `Boolean` constructor :)